File: backend/app/schemas/team.py
# backend/app/schemas/team.py
from pydantic import BaseModel
from typing import Optional, List
from uuid import UUID
from datetime import datetime


class TeamMemberCreate(BaseModel):
    full_name: str
    email: Optional[str] = None
    phone: Optional[str] = None
    organization: Optional[str] = None
    is_captain: bool = False


class TeamMemberUpdate(BaseModel):
    full_name: Optional[str] = None
    email: Optional[str] = None
    phone: Optional[str] = None
    organization: Optional[str] = None
    is_captain: Optional[bool] = None


class TeamMemberResponse(BaseModel):
    id: UUID
    team_id: UUID
    full_name: str
    email: Optional[str] = None
    phone: Optional[str] = None
    organization: Optional[str] = None
    is_captain: bool
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True


class TeamCreate(BaseModel):
    name: str
    captain_name: str
    contact_email: Optional[str] = None
    contact_phone: Optional[str] = None
    project_title: str
    description: Optional[str] = None


class TeamUpdate(BaseModel):
    name: Optional[str] = None
    captain_name: Optional[str] = None
    contact_email: Optional[str] = None
    contact_phone: Optional[str] = None
    project_title: Optional[str] = None
    description: Optional[str] = None


class TeamResponse(BaseModel):
    id: UUID
    hackathon_id: UUID
    name: str
    captain_name: str
    contact_email: Optional[str] = None
    contact_phone: Optional[str] = None
    project_title: str
    description: Optional[str] = None
    members_count: int = 0
    evaluation_status: Optional[str] = None
    final_score: Optional[float] = None
    place: Optional[int] = None
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True


class AssignedExpertInfo(BaseModel):
    expert_id: UUID
    expert_name: str
    evaluation_status: Optional[str] = None
    evaluation_id: Optional[UUID] = None


class TeamDetailResponse(BaseModel):
    id: UUID
    hackathon_id: UUID
    name: str
    captain_name: str
    contact_email: Optional[str] = None
    contact_phone: Optional[str] = None
    project_title: str
    description: Optional[str] = None
    members: List[TeamMemberResponse] = []
    assigned_experts: Optional[List[AssignedExpertInfo]] = []  # Сделано опциональным
    evaluation_status: Optional[str] = None  # Может быть None
    final_score: Optional[float] = None
    place: Optional[int] = None
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True


class TeamListResponse(BaseModel):
    items: List[TeamResponse]
    page: int
    page_size: int
    total: int

File: frontend/lib/models/team.dart
class Team {
  final String id;
  final String name;
  final String captainName;
  final String? contactEmail;
  final String? contactPhone;
  final String projectTitle;
  final String? description;
  final int membersCount;
  final String evaluationStatus;
  final double? finalScore;
  final int? place;

  Team({
    required this.id,
    required this.name,
    required this.captainName,
    this.contactEmail,
    this.contactPhone,
    required this.projectTitle,
    this.description,
    required this.membersCount,
    required this.evaluationStatus,
    this.finalScore,
    this.place,
  });

  factory Team.fromJson(Map<String, dynamic> json) {
    return Team(
      id: json['id'],
      name: json['name'],
      captainName: json['captain_name'],
      contactEmail: json['contact_email'],
      contactPhone: json['contact_phone'],
      projectTitle: json['project_title'],
      description: json['description'],
      membersCount: json['members_count'] ?? 0,
      evaluationStatus: json['evaluation_status'] ?? 'not_started',
      finalScore: (json['final_score'] as num?)?.toDouble(),
      place: json['place'],
    );
  }
}

class TeamMember {
  final String id;
  final String fullName;
  final String? email;
  final String? phone;
  final String? organization;
  final bool isCaptain;

  TeamMember({
    required this.id,
    required this.fullName,
    this.email,
    this.phone,
    this.organization,
    required this.isCaptain,
  });

  factory TeamMember.fromJson(Map<String, dynamic> json) {
    return TeamMember(
      id: json['id'],
      fullName: json['full_name'],
      email: json['email'],
      phone: json['phone'],
      organization: json['organization'],
      isCaptain: json['is_captain'] ?? false,
    );
  }
}
